jbpm-console-ng-process-runtime/jbpm-console-ng-process-runtime-client/src/main/java/org/jbpm/console/ng/pr/client/editors/instance/list/variables/dash/DataSetProcessInstanceWithVariablesListViewImpl.java

```diff
@@ -372,23 +373,57 @@ public String getValue( ProcessInstanceSummary object ) {
     }
 
     public void initExtraButtons( final ExtendedPagedTable<ProcessInstanceSummary> extendedPagedTable ) {
+        final Button selectAllButton = GWT.create(Button.class);
+        selectAllButton.setText(constants.SelectAll());
+        selectAllButton.setIcon(IconType.SQUARE_O);
+        selectAllButton.setTitle(constants.SelectAllTooltip());
+        selectAllButton.setIcon(IconType.SQUARE_O);
+        selectAllButton.addClickHandler(new ClickHandler() {
+            @Override
+            public void onClick(ClickEvent event) {
+                selectedProcessInstances.clear();
+                if (IconType.SQUARE_O.equals(selectAllButton.getIcon())) {
+                    selectedProcessInstances.addAll(presenter.getDisplayedProcessInstances());
+                    presenter.refreshGrid();
```

Comment:
Why exactly do we need to refresh the data? It also seems that we're doing it twice at L396.